src/seedu/addressbook/data/person/Attendance.java

```diff
@@ -1,30 +1,40 @@
 package seedu.addressbook.data.person;
 
+import java.text.SimpleDateFormat;
+import java.util.Date;
 import java.util.HashMap;
 import java.util.Map;
 
 /**
- *  Represents a map that links dates(kay) to attendance(value)
+ * Represents a Person's attendance in the address book.
  */
 
 public class Attendance implements Printable {
 
+    /** Represents a map that links dates(kay) to attendance(value)*/
     private Map<String, Boolean> attendanceMap = new HashMap<>();
-    /**
-    // Method to add attendance
-     */
-    public void addAttendance(String currentDate, String isPresent) {
-        boolean isPresentNow = "1".equals(isPresent);
-        attendanceMap.put(currentDate, isPresentNow);
+
+    /** Method to add attendance */
+    public void addAttendance(String date, Boolean isPresent) {
+        if ("0".equals(date)) { //PMD 3.3
+            String todayDate = new SimpleDateFormat("dd-MM-yyyy").format(new Date());
```

Comment:
What happens if the string date is not in dd-MM-yyyy format?

Reply:
Neh, the dd-MM-yyyy is the generated string base on "today's" date

Reply:
if its not "0"

Reply:
So, you enter the date as a string, or choose to enter '0' instead, of which it will autogenerate the date for you

Reply:
Correct, but theres no check if the date string is correct if date not "0"